src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
```

Comment:
Dima, also is my concern in the comment above valid? Can a pending resource be delayed unnecessary because we haven't encounter another non-fully parsed element yet? 

If not I can't think of a negative out-of-order building. So we can do this if there isn't a concern of delaying some resources.